### PR TITLE
Lambda catalonia adapter

### LINKED ISSUE #709: New data: Spain Catalonia
Found this datasource for Catalonia by the Catalonian department of Department of Territory and Sustainability: 
http://mediambient.gencat.cat/ca/05_ambits_dactuacio/atmosfera/qualitat_de_laire/vols-saber-que-respires/visor-de-dades/
Is it usable?

### PR DESCRIPTION
**Pull request by majesticio**
adapted from PR #711 
credit to @magsyg
closes #711 
closes #709

### CODE REVIEW

File: src/adapters/catalonia.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+ * This code is responsible for implementing all methods related to fetching
+ * and returning data for the Catalonian data sources. credit to @magsyg
+ * based off of openaq-fetch PR #711 
+ */
+
+'use strict';
+
+import { REQUEST_TIMEOUT } from '../lib/constants.js';
+import { default as baseRequest } from 'request';
+import { default as moment } from 'moment-timezone';
```
Comment (russbiggs) [inline src/adapters/catalonia.js:11]: Lets go ahead and change new stuff to luxon moving forward

File: src/sources/es.json [hunk 2]
```diff
@@ -23,5 +23,17 @@
             "info@openaq.org"
         ],
         "active": true
+    },
+    {
+        "url": "https://analisi.transparenciacatalunya.cat/resource/tasf-thgu.json",
+        "name": "Spain Catalonia",
```
Comment (russbiggs) [inline src/sources/es.json:29]: Rename to reflect the provider better

File: src/adapters/catalonia.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      String(param).localeCompare('cl2') !== 0 &&
+      String(param).localeCompare('hg') !== 0 &&
+      String(param).localeCompare('pm1') !== 0) {
+
+      const template = {
+        location: ('nom_estaci' in item) ? item.nom_estaci : item.municipi,
+        city: item.municipi,
+        parameter: param,
+        coordinates: {
+          latitude: Number(item.latitud),
```
Comment (russbiggs) [inline src/adapters/catalonia.js:78]: lets do parseFloat instead of Number

File: src/adapters/catalonia.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      };
+
+      // Loop through all hours and check if there is any data for that hour on that day
+      for (let i = 1; i < 25; i++) {
+        dateMoment = moment(dateMoment).add(1, 'hours').format('YYYY-MM-DD HH:mm');
+        dateMoment = moment.tz(dateMoment, 'YYYY-MM-DD HH:mm', 'Europe/Madrid');
+        let valueKey = (i < 10) ? ('h0' + i.toString()) : ('h' + i.toString());
+        if (valueKey in item) {
+          let temp = Object.assign({
+            value: Number(item[valueKey]),
```
Comment (russbiggs) [inline src/adapters/catalonia.js:93]: parseFloat

**Review by russbiggs — CHANGES_REQUESTED** [review]

**Comment by russbiggs** [comment]
@majesticio lets do a wiki entry for this source before merging

**Comment by majesticio** [comment]
Catalonia wiki has been created
https://github.com/openaq/openaq-fetch/wiki/Catalonia-Medi-Ambient-i-Sostenibilitat

**Review by russbiggs — APPROVED** [review]

File: src/adapters/catalonia.js (hunk 1 above)
Comment (russbiggs) [inline src/adapters/catalonia.js:11]: This still needs to be updated to the luxon

**Review by russbiggs — CHANGES_REQUESTED** [review]

File: src/adapters/catalonia.js (hunk 1 above)
Comment (majesticio) [reply src/adapters/catalonia.js:11]: This is one that I was not able to reproduce what magsyg did with moment in luxon

**Comment by majesticio** [comment]
included params 'no', 'pm1' in commit 77b6f22

File: src/adapters/catalonia.js (hunk 1 above)
Comment (majesticio) [reply src/adapters/catalonia.js]: figured it out, added 'no, nox, pm1' to enum, acceptableParameters

**Pull request merged**